adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -873,10 +865,11 @@
         function checkLoginIframe() {
             var promise = createPromise();
 
-            if (loginIframe.iframe && loginIframe.iframeOrigin) {
+            if (loginIframe.iframe && loginIframe.iframeOrigin && loginIframe.callbackList.length !== 0) {
                 var msg = {};
                 msg.callbackId = createCallbackId();
-                loginIframe.callbackMap[msg.callbackId] = promise;
+                msg.sessionId = kc.sessionId;
+                loginIframe.callbackList.push(promise);
```

Comment:
But that won't work the first time , since we add the promise to the list , it never sends a message and the the callback will never be called to resolve the promise.